### PR TITLE
Add checkbox

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: components/atoms/checkbox/checkbox.core.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  return (
+    <NextUiCheckbox
+      classNames={{
+        base: cn(slots.base(), classNames?.base),
+        label: cn(slots.label(), classNames?.label),
+        wrapper: cn(slots.wrapper(), classNames?.wrapper),
+        icon: cn(slots.icon(), classNames?.icon),
+      }}
+      isDisabled={disabled}
+      disabled={disabled}
```
Comment (haydencleary) [inline components/atoms/checkbox/checkbox.core.tsx:29]: Need both ?

File: components/atoms/checkbox/checkbox.types.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import { CheckboxCoreVariants } from "./checkbox.variants";
+
+type Variants = VariantProps<typeof CheckboxCoreVariants>;
+type classNames = Partial<typeof CheckboxCoreVariants["slots"]>;
+export interface TCheckboxProps extends Variants {
+  classNames?: classNames;
+  onChange?: (checked: boolean) => void;
+  value?: boolean;
+  disabled?: boolean;
```
Comment (haydencleary) [inline components/atoms/checkbox/checkbox.types.ts:11]: I think `isDisabled` is better for state

File: components/atoms/checkbox/variants/checkbox-default.tsx [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { CheckboxCore } from "../checkbox.core";
+import { TCheckboxProps } from "../checkbox.types";
+import { ElementType } from "react";
+
+export function Checkbox<C extends ElementType = "div">({
+  ...props
+}: TCheckboxProps<C>) {
+  return <CheckboxCore {...props} classNames={{}} />;
```
Comment (haydencleary) [inline components/atoms/checkbox/variants/checkbox-default.tsx:8]:
```suggestion
  return <CheckboxCore {...props} />;
```

**Review by haydencleary — CHANGES_REQUESTED** [review]

File: components/atoms/checkbox/checkbox.core.tsx (hunk 1 above)
Comment (alexbeno) [reply components/atoms/checkbox/checkbox.core.tsx:29]: next ui api :/

**Review by haydencleary — APPROVED** [review]

**Pull request merged**